lib/commands/general.js

```diff
@@ -13,24 +13,26 @@ commands.getWindowSize = async function getWindowSize () {
   }
   // workaround for https://github.com/microsoft/WinAppDriver/issues/1104
   log.info('Cannot retrieve window size from WinAppDriver. ' +
-    'Falling back to WMIC usage');
-  const {stdout} = await exec('wmic.exe', [
-    'path', 'Win32_VideoController',
-    'get', 'VideoModeDescription,CurrentVerticalResolution,CurrentHorizontalResolution',
-    '/format:value',
-  ]);
-  log.debug(stdout);
-  const widthMatch = /CurrentHorizontalResolution=(\d+)/.exec(stdout);
-  const heightMatch = /CurrentVerticalResolution=(\d+)/.exec(stdout);
-  if (!widthMatch || !heightMatch) {
+      'Falling back to Windows Forms to calculate dimensions');
+  const width = await getDimension('Width');
+  const height = await getDimension('Height');
+  if (isNaN(width) || isNaN(height)) {
     throw new Error('Cannot retrieve the screen size. Check the server log for more details');
   }
   return {
-    width: parseInt(widthMatch[1], 10),
-    height: parseInt(heightMatch[1], 10),
+    width,
+    height,
   };
 };
 
+async function getDimension (dimensionType) {
+  const {stdout} = await exec('powershell', [
+    '-command', `Add-Type -AssemblyName System.Windows.Forms;[System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Size.${dimensionType}`,
```

Comment:
this command could be changed to 

```
    '-command', `Add-Type -AssemblyName System.Windows.Forms;[System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Size`,
```

to get both dimensions. Then the regexp to parse the output would looks like

```
const match = /^\s*(True|False)\s+(\d+)\s+(\d+)/.exec(stdout)
if (!match) {
  throw...
}
const width = parseInt(match[2], 10);
const height = parseInt(match[3], 10);
```

Reply:
Thanks!  Saved me time constructing a regex (not my strongest area!)